actors/common/src/main/java/org/sunbird/common/quartz/scheduler/OnDemandShadowManager.java

```diff
@@ -0,0 +1,216 @@
+   JobDetail migrateShadowUserJob =
+           JobBuilder.newJob(ShadowUserMigrationScheduler.class)
+                   .requestRecovery(true)
+                   .withDescription("Scheduler for migrating shadow user ")
+                   .withIdentity("migrateShadowUserScheduler", identifier)
+                   .build();
+   Trigger migrateShadowUserTrigger =
+           TriggerBuilder.newTrigger()
+                   .withIdentity("migrateShadowUserTrigger", identifier).startNow()
+                   .withSchedule(simpleSchedule().withIntervalInSeconds(1).withRepeatCount(0))
+                   .build();
+   try {
+    if (scheduler.checkExists(migrateShadowUserJob.getKey())) {
+     scheduler.deleteJob(migrateShadowUserJob.getKey());
+    }
+    scheduler.scheduleJob(migrateShadowUserJob, migrateShadowUserTrigger);
+    scheduler.start();
+    ProjectLogger.log(
+            "OnDemandShadowManager:scheduleShadowUserOnDemand:scheduler ended",
+            LoggerEnum.INFO.name());
+   } catch (Exception e) {
+    ProjectLogger.log(
+            "OnDemandShadowManager:scheduleShadowUserOnDemand Error occurred " + e.getMessage(),
+            LoggerEnum.ERROR.name());
+   }
+  }
+
+  /**
+   * This method will do the Quartz scheduler set up in cluster mode.
+   *
+   * @return Properties
+   * @throws IOException
+   */
+  public Properties setUpClusterMode() throws IOException {
+   Properties configProp = new Properties();
+   InputStream in = this.getClass().getClassLoader().getResourceAsStream(FILE);
+   String host = System.getenv(JsonKey.SUNBIRD_PG_HOST);
+   String port = System.getenv(JsonKey.SUNBIRD_PG_PORT);
+   String db = System.getenv(JsonKey.SUNBIRD_PG_DB);
+   String username = System.getenv(JsonKey.SUNBIRD_PG_USER);
+   String password = System.getenv(JsonKey.SUNBIRD_PG_PASSWORD);
+   if (!StringUtils.isBlank(host)
+           && !StringUtils.isBlank(port)
+           && !StringUtils.isBlank(db)
+           && !StringUtils.isBlank(username)
+           && !StringUtils.isBlank(password)) {
+    ProjectLogger.log(
+            "Taking Postgres value from Environment variable...", LoggerEnum.INFO.name());
+    configProp.load(in);
+    configProp.put(
+            "org.quartz.dataSource.MySqlDS.URL", "jdbc:postgresql://" + host + ":" + port + "/" + db);
+    configProp.put("org.quartz.dataSource.MySqlDS.user", username);
+    configProp.put("org.quartz.dataSource.MySqlDS.password", password);
+    ProjectLogger.log(
+            "OnDemandShadowManager:setUpClusterMode: Connection is established from environment variable",
+            LoggerEnum.INFO);
+   } else {
+    ProjectLogger.log(
+            "OnDemandShadowManager:setUpClusterMode: Environment variable is not set for postgres SQl.",
+            LoggerEnum.INFO.name());
+    configProp = null;
+   }
+   return configProp;
+  }
+
+  public static org.sunbird.common.quartz.scheduler.OnDemandShadowManager getInstance() {
+   if (OnDemandShadowManager != null) {
```

Comment:
can be better:
if (OnDemandShadowManager == null) {
OnDemandShadowManager = new org.sunbird.common.quartz.scheduler.OnDemandShadowManager();
}
return OnDemandShadowManager;

Reply:
done